test/KrakenIntegrationTest.cpp

```diff
@@ -74,24 +82,48 @@ TEST_F(KrakenIntegrationTest, VerifyCommunication) {
    auto spear = kraken.SetLocation(queue);
    EXPECT_EQ(spear, Kraken::Spear::IMPALED) << "spear: " << static_cast<int>(spear);
 
-   size_t tooHighValue = 100;
-   std::shared_ptr<std::atomic<size_t>> allReceived {new std::atomic<size_t>{tooHighValue}};
+   std::shared_ptr<std::atomic<size_t>> expectedToReceive {new std::atomic<size_t>{0}};
    std::shared_ptr<Harpoon> harpoon = std::make_shared<Harpoon>();
-   harpoon->Aim(queue);
+   auto spearResult = harpoon->Aim(queue);
+   ASSERT_EQ(spearResult, Harpoon::Spear::IMPALED);
    harpoon->MaxWaitInMs(1000);
 
    // RECEIVER
    using HarpoonReceived = std::vector<Kraken::Chunks>;
-   std::future<HarpoonReceived> willReceive = std::async(std::launch::async,  [allReceived, harpoon] {
+   std::future<HarpoonReceived> willReceive = std::async(std::launch::async,  [expectedToReceive, harpoon] {
```

Comment:
I've focused on making the flow of the data much clearer. For example the last data to receive Line: 121